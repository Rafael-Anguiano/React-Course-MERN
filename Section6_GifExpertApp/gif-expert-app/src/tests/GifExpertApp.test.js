import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import GifExpertApp from '../GifExpertApp'

describe('Test in GifExpertApp component', () => {
    
    test('should ', () => {
        const wrapper = shallow(<GifExpertApp/>)
        expect(wrapper).toMatchSnapshot()
    })
    
    test('should show a list of categories', () => {
        const categories = ['HxH', 'Naruto']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length ).toBe(categories.length)
    })
    
    

})