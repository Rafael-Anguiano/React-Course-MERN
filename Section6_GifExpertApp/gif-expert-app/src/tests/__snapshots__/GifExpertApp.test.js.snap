// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifExpertApp component should  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test in GifExpertApp component should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="HxH"
      key="HxH"
    />
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
  </ol>
</Fragment>
`;
